packages/video_player/video_player_web/lib/src/video_player.dart

```diff
@@ -202,9 +203,35 @@ class VideoPlayer {
     return Duration(milliseconds: (_videoElement.currentTime * 1000).round());
   }
 
+  /// Sets options
+  Future<void> setOptions(VideoPlayerWebOptions options) async {
```

Comment:
Can this method be called multiple times on the same _videoElement? Does this method need to support both cases, setting and unsetting options?

Imagine that this gets called with `allowContextMenu=false` once, and then `allowContextMenu=true`

Reply:
(Since there's no mechanism in the code that prevents a core plugin from calling `setWebOptions` multiple times, I think this method should work if called multiple times in succession, and be able to set/remove values).

Reply:
As per #3259, the use case is `video_player/lib/video_player.dart` calls `setWebOptions` once in `initialize`, however there is nothing in place to stop this method being called multiple times. I've added a check to de-initialize attributes when required.